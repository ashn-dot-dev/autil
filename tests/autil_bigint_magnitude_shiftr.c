#define AUTIL_IMPLEMENTATION
#include "../autil.h"

int
main(void)
{
    struct autil_bigint* x = autil_bigint_new_cstr("0xff0000");
    autil_bigint_dump(x);

    autil_bigint_magnitude_shiftr(x, 0);
    autil_bigint_dump(x);

    autil_bigint_magnitude_shiftr(x, 4);
    autil_bigint_dump(x);

    autil_bigint_magnitude_shiftr(x, 7);
    autil_bigint_dump(x);

    autil_bigint_magnitude_shiftr(x, 1);
    autil_bigint_dump(x);

    autil_bigint_magnitude_shiftr(x, 32);
    autil_bigint_dump(x);

    autil_bigint_magnitude_shiftr(x, 4096);
    autil_bigint_dump(x);

    struct autil_bigint* const y = autil_bigint_new_cstr("-0b10");
    autil_bigint_dump(y);
    autil_bigint_magnitude_shiftr(y, 1);
    autil_bigint_dump(y);
    autil_bigint_magnitude_shiftr(y, 1);
    autil_bigint_dump(y);

    autil_bigint_del(x);
    autil_bigint_del(y);
    return EXIT_SUCCESS;
}
