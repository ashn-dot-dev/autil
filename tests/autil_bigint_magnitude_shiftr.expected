SIGN: '+', COUNT: 3, LIMBS: [0x00, 0x00, 0xFF]
SIGN: '+', COUNT: 3, LIMBS: [0x00, 0x00, 0xFF]
SIGN: '+', COUNT: 3, LIMBS: [0x00, 0xF0, 0x0F]
SIGN: '+', COUNT: 2, LIMBS: [0xE0, 0x1F]
SIGN: '+', COUNT: 2, LIMBS: [0xF0, 0x0F]
SIGN: '0', COUNT: 0, LIMBS: []
SIGN: '0', COUNT: 0, LIMBS: []
SIGN: '-', COUNT: 1, LIMBS: [0x02]
SIGN: '-', COUNT: 1, LIMBS: [0x01]
SIGN: '0', COUNT: 0, LIMBS: []
